pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -36,8 +37,7 @@ import (
 )
 
 func TestCompatibility_v1_Scheduler(t *testing.T) {
-	snapshot := scheduler.RegisteredPredicatesAndPrioritiesSnapshot()
-	defer scheduler.ApplyPredicatesAndPriorities(snapshot)
+	defer algorithmprovider.ApplyFeatureGates()()
```

Comment:
Can we add some examples to check that if we intentionally enable/disable one feature gate, it works as expected?

- Alpha example: Enable EvenPodsSpread and check the result.
- Beta example: Disable a Beta feature gate and check the result.